app/components/provider_interface/safeguarding_declaration_component.rb

```diff
@@ -2,13 +2,43 @@ module ProviderInterface
   class SafeguardingDeclarationComponent < ViewComponent::Base
     include ViewHelper
 
-    attr_reader :message
+    attr_reader :application_choice, :current_provider_user, :details
 
-    def initialize(application_form:)
-      @message = SafeguardingStatus.new(
-        status: application_form.safeguarding_issues_status,
+    def initialize(application_choice:, current_provider_user:)
+      @application_choice = application_choice
+      @current_provider_user = current_provider_user
+    end
+
+    def message
+      set_details if safeguarding_issues_declared? && display_safeguarding_information?
+
+      SafeguardingStatus.new(
+        status: status,
         i18n_key: 'provider_interface.safeguarding_declaration_component',
       ).message
     end
+
+    def display_safeguarding_information?
+      ProviderAuthorisation.new(actor: current_provider_user)
+        .can_view_safeguarding_information?(course: application_choice.course)
+    end
+
+  private
+
+    def status
+      if safeguarding_issues_declared? && !display_safeguarding_information?
+        'has_safeguarding_issues_to_declare_no_permissions'
+      else
+        application_choice.application_form.safeguarding_issues_status
+      end
+    end
+
+    def safeguarding_issues_declared?
+      application_choice.application_form.safeguarding_issues_status == 'has_safeguarding_issues_to_declare'
+    end
+
+    def set_details
+      @details = application_choice.application_form.safeguarding_issues
+    end
```

Comment:
Is `@details` used? I can't find it in the HTML.

Reply:
It's used here: https://github.com/DFE-Digital/apply-for-teacher-training/blob/3f122fb59ad13134c2cb70b532d38268ab6b4ea4/app/components/provider_interface/safeguarding_declaration_component.html.erb#L10

Reply:
Ah, it's exposed via `attr_reader`. Should we make this into a method so it's similar to how we use `message`?

Reply:
I was a bit unsure about this bit and that I'm setting the details within message:
```
def message
      set_details if safeguarding_issues_declared? && display_safeguarding_information?
```
but then I only wanted to assign whatever candidate disclosed to `details` if the provider_user can view it. 
I guess we have `<% if display_safeguarding_information? %>` in the view so it should be fine.

Should I go with public
```
   def details
      application_choice.application_form.safeguarding_issues
    end
```
and remove `set_details if ...` from #message?

Reply:
Yeah, that sounds correct. It removes some duplication as well!